importing the aws-sdk results in a module not found error

## :bug: Bug Report

### Affected Languages
<!--
Check the box (with an X) for any language runtime that you know is affected by
the reported bug. If you're uncertain whether a language is affected or not,
please leave the bux un-checked.
-->
- [X] `TypeScript` or `Javascript`
- [ ] `Python`
- [ ] `Java`
- [ ] .NET (`C#`, `F#`, ...)

### General Information
* **JSII Version:** 1.4.1 (build 0b09d7b), typescript 3.8.3
* **Platform:**     Linux ip-172-31-23-62 4.14.173-106.229.amzn1.x86_64 #1 SMP Wed Apr 1 19:40:12 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

### What is the problem?
running `npx tsc` works correctly and. generates the expected files, but running `npx jsii` or `npm run build` (which references `jsii`) fails.
my `index.ts` file:

````typescript
import {
    ClientRequestToken,
    LogicalResourceId,
    NextToken,
} from 'aws-sdk/clients/cloudformation';


export class HelloJsii {
  public sayHello(name: string) {
    return `Hello, ${name}!`;
  }
}
````

### Verbose Log
````text
CrossAccountRole-Isengard:~/environment/jsii-rpdk $ npm run build --verbose
npm info it worked if it ends with ok
npm verb cli [ '/home/ec2-user/.nvm/versions/node/v10.20.1/bin/node',
npm verb cli   '/home/ec2-user/.nvm/versions/node/v10.20.1/bin/npm',
npm verb cli   'run',
npm verb cli   'build',
npm verb cli   '--verbose' ]
npm info using npm@6.14.4
npm info using node@v10.20.1
npm verb run-script [ 'prebuild', 'build', 'postbuild' ]
npm info lifecycle jsii-rpdk@1.0.0~prebuild: jsii-rpdk@1.0.0
npm info lifecycle jsii-rpdk@1.0.0~build: jsii-rpdk@1.0.0

> jsii-rpdk@1.0.0 build /home/ec2-user/environment/jsii-rpdk
> jsii

Error: Error: Unable to locate module: aws-sdk
    at _tryResolveAssembly (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/lib/project-info.js:200:15)
    at _loadDependencies (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/lib/project-info.js:125:21)
    at Object.loadProjectInfo (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/lib/project-info.js:63:32)
    at /home/ec2-user/environment/jsii-rpdk/node_modules/jsii/bin/jsii.js:45:46
    at Object.<anonymous> (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/bin/jsii.js:62:3)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
npm verb lifecycle jsii-rpdk@1.0.0~build: unsafe-perm in lifecycle true
npm verb lifecycle jsii-rpdk@1.0.0~build: PATH: /home/ec2-user/.nvm/versions/node/v10.20.1/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/ec2-user/environment/jsii-rpdk/node_modules/.bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/ec2-user/.nvm/versions/node/v10.20.1/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3@global/bin:/home/ec2-user/.rvm/rubies/ruby-2.6.3/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/ec2-user/.rvm/gems/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3@global/bin:/home/ec2-user/.rvm/rubies/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3@global/bin:/home/ec2-user/.rvm/rubies/ruby-2.6.3/bin:/usr/local/bin:/bin:/usr/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/home/ec2-user/.rvm/bin:/usr/local/sbin:/usr/sbin:/sbin:/opt/aws/bin:/usr/local/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/home/ec2-user/.rvm/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/opt/aws/bin:/usr/local/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/home/ec2-user/.rvm/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin
npm verb lifecycle jsii-rpdk@1.0.0~build: CWD: /home/ec2-user/environment/jsii-rpdk
npm info lifecycle jsii-rpdk@1.0.0~build: Failed to exec build script
npm verb stack Error: jsii-rpdk@1.0.0 build: `jsii`
npm verb stack Exit status 255
npm verb stack     at EventEmitter.<anonymous> (/home/ec2-user/.nvm/versions/node/v10.20.1/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
npm verb stack     at EventEmitter.emit (events.js:198:13)
npm verb stack     at ChildProcess.<anonymous> (/home/ec2-user/.nvm/versions/node/v10.20.1/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
npm verb stack     at ChildProcess.emit (events.js:198:13)
npm verb stack     at maybeClose (internal/child_process.js:982:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
npm verb pkgid jsii-rpdk@1.0.0
npm verb cwd /home/ec2-user/environment/jsii-rpdk
npm verb Linux 4.14.173-106.229.amzn1.x86_64
npm verb argv "/home/ec2-user/.nvm/versions/node/v10.20.1/bin/node" "/home/ec2-user/.nvm/versions/node/v10.20.1/bin/npm" "run" "build" "--verbose"
npm verb node v10.20.1
npm verb npm  v6.14.4
npm ERR! code ELIFECYCLE
npm ERR! errno 255
npm ERR! jsii-rpdk@1.0.0 build: `jsii`
npm ERR! Exit status 255
npm ERR! 
npm ERR! Failed at the jsii-rpdk@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm verb exit [ 255, true ]
npm timing npm Completed in 511ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ec2-user/.npm/_logs/2020-05-07T01_11_26_116Z-debug.log
````

My node_module:
````text
CrossAccountRole-Isengard:~/environment/jsii-rpdk $ ls node_modules/aws-sdk/
apis        buildspec.yml  clients  dist-tools   global.js   index.js  LICENSE.txt  package.json     README.md  vendor
browser.js  CHANGELOG.md   dist     global.d.ts  index.d.ts  lib       NOTICE.txt   react-native.js  scripts
````

My `package.json`:
````json
  "devDependencies": {
    "@types/node": "^13.13.5",
    "aws-sdk": "^2.671.0",
    "jsii": "^1.4.1",
    "jsii-pacmak": "^1.4.1"
  },
````

## Edit
I tried again with the latest `1.5.0` release and the error is still happening.

`package.json`
````json
  "devDependencies": {
    "@types/node": "^13.13.5",
    "aws-sdk": "^2.671.0",
    "jsii": "1.5.0",
    "jsii-pacmak": "1.5.0"
  },
````

`index.ts`
````typescript
import AWS from 'aws-sdk';

export class HelloJsii {
  public sayHello(name: string) {
    return `Hello, ${name}!`;
  }
}
````

`--verbose` jsii output
````text
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   '/home/ec2-user/.nvm/versions/node/v12.16.3/bin/node',
npm verb cli   '/home/ec2-user/.nvm/versions/node/v12.16.3/bin/npm',
npm verb cli   'run',
npm verb cli   'build',
npm verb cli   '--verbose'
npm verb cli ]
npm info using npm@6.14.4
npm info using node@v12.16.3
npm verb run-script [ 'prebuild', 'build', 'postbuild' ]
npm info lifecycle jsii-rpdk@1.0.0~prebuild: jsii-rpdk@1.0.0
npm info lifecycle jsii-rpdk@1.0.0~build: jsii-rpdk@1.0.0

> jsii-rpdk@1.0.0 build /home/ec2-user/environment/jsii-rpdk
> jsii

Error: Error: Unable to locate module: aws-sdk
    at _tryResolveAssembly (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/lib/project-info.js:200:15)
    at _loadDependencies (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/lib/project-info.js:125:21)
    at Object.loadProjectInfo (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/lib/project-info.js:63:32)
    at /home/ec2-user/environment/jsii-rpdk/node_modules/jsii/bin/jsii.js:45:46
    at Object.<anonymous> (/home/ec2-user/environment/jsii-rpdk/node_modules/jsii/bin/jsii.js:62:3)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Module.require (internal/modules/cjs/loader.js:1019:19)
npm verb lifecycle jsii-rpdk@1.0.0~build: unsafe-perm in lifecycle true
npm verb lifecycle jsii-rpdk@1.0.0~build: PATH: /home/ec2-user/.nvm/versions/node/v12.16.3/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/ec2-user/environment/jsii-rpdk/node_modules/.bin:/home/ec2-user/.nvm/versions/node/v12.16.3/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/ec2-user/.rvm/gems/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3@global/bin:/home/ec2-user/.rvm/rubies/ruby-2.6.3/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/ec2-user/.rvm/gems/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3@global/bin:/home/ec2-user/.rvm/rubies/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3/bin:/home/ec2-user/.rvm/gems/ruby-2.6.3@global/bin:/home/ec2-user/.rvm/rubies/ruby-2.6.3/bin:/usr/local/bin:/bin:/usr/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/home/ec2-user/.rvm/bin:/usr/local/sbin:/usr/sbin:/sbin:/opt/aws/bin:/usr/local/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/home/ec2-user/.rvm/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/opt/aws/bin:/usr/local/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin:/home/ec2-user/.rvm/bin:/home/ec2-user/.local/bin:/home/ec2-user/bin
npm verb lifecycle jsii-rpdk@1.0.0~build: CWD: /home/ec2-user/environment/jsii-rpdk
npm info lifecycle jsii-rpdk@1.0.0~build: Failed to exec build script
npm verb stack Error: jsii-rpdk@1.0.0 build: `jsii`
npm verb stack Exit status 255
npm verb stack     at EventEmitter.<anonymous> (/home/ec2-user/.nvm/versions/node/v12.16.3/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
npm verb stack     at EventEmitter.emit (events.js:310:20)
npm verb stack     at ChildProcess.<anonymous> (/home/ec2-user/.nvm/versions/node/v12.16.3/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
npm verb stack     at ChildProcess.emit (events.js:310:20)
npm verb stack     at maybeClose (internal/child_process.js:1021:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
npm verb pkgid jsii-rpdk@1.0.0
npm verb cwd /home/ec2-user/environment/jsii-rpdk
npm verb Linux 4.14.173-106.229.amzn1.x86_64
npm verb argv "/home/ec2-user/.nvm/versions/node/v12.16.3/bin/node" "/home/ec2-user/.nvm/versions/node/v12.16.3/bin/npm" "run" "build" "--verbose"
npm verb node v12.16.3
npm verb npm  v6.14.4
npm ERR! code ELIFECYCLE
npm ERR! errno 255
npm ERR! jsii-rpdk@1.0.0 build: `jsii`
npm ERR! Exit status 255
npm ERR! 
npm ERR! Failed at the jsii-rpdk@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm verb exit [ 255, true ]
npm timing npm Completed in 419ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ec2-user/.npm/_logs/2020-05-07T19_34_44_557Z-debug.log
````
<!--
* If your issue is with `jsii`: please provide the output of running `jsii` with
  the `--verbose` option.
* If your issue is with `jsii-pacmak`: please provide the output of running
  `jsii-pacmak` with the `--verbose` option.
* If your issue is when running generated code: please provide the output of
  running your program with the `JSII_DEBUG=1` environment variable being set.

If the output is very large, you may store it in a [gist][1]. In any case,
please review the contents of the log you are about to submit and edit any
confidential information out.

[1]: https://gist.github.com
-->

--- a/packages/jsii/lib/project-info.ts
+++ b/packages/jsii/lib/project-info.ts
@@ -253,7 +253,7 @@ function _tryResolveAssembly(mod: string, localPackage: string | undefined, sear
     const paths = [searchPath, path.join(searchPath, 'node_modules')];
     return require.resolve(path.join(mod, '.jsii'), { paths });
   } catch {
-    throw new Error(`Unable to locate module: ${mod}`);
+    throw new Error(`Unable to locate jsii assembly for "${mod}". If this module is not jsii-enabled, it must be bundled.`);
   }
 }
 

--- a/packages/jsii/lib/project-info.ts
+++ b/packages/jsii/lib/project-info.ts
@@ -253,7 +253,7 @@ function _tryResolveAssembly(mod: string, localPackage: string | undefined, sear
     const paths = [searchPath, path.join(searchPath, 'node_modules')];
     return require.resolve(path.join(mod, '.jsii'), { paths });
   } catch {
-    throw new Error(`Unable to locate jsii assembly for "${mod}". If this module is not jsii-enabled, it must be bundled.`);
+    throw new Error(`Unable to locate jsii assembly for "${mod}". If this module is not jsii-enabled, it must also be declared under bundledDependencies.`);
   }
 }
 

